packages/typespec-client-generator-core/test/decorators.test.ts

```diff
@@ -3284,7 +3290,7 @@ describe("typespec-client-generator-core: decorators", () => {
         `
       );
       const sdkPackage = runner.context.experimental_sdkPackage;
-      strictEqual(sdkPackage.clients.length, 2);
+      strictEqual(sdkPackage.clients.length, 1);
```

Comment:
Is the clients.length 2 before because we treated namespace Versioning and interface InterfaceV2 as the sub clients ?  
I am curious what the expected client constructor for IntefaceV2 is now that it's a sub client for Versioning?

Reply:
yes. previously, we will expose all the sub clients. for now, versioning client has a method `getInterfaceV2` which response is the interfacev2 client. since the client's initialization of interfacev2 is private, it should only be initialized with its parent client.